src/components/FilterBar/FilterDisplay/Inputs/DateInputRelative.react.js

```diff
@@ -0,0 +1,139 @@
+export class DateInputRelative extends React.Component {
+  constructor(props) {
+    super(props);
+    this.props.dateFormat = 'DD/MM/YYYY'
+    this.state = { value: this.props.value || { from: null, to: null, value: null } };
+  }
+
+  onDateInputChange(event) {
+    var newValue = this.state.value;
+
+    if(event.type === "dp") {
+      newValue[event.target.querySelector("input").getAttribute("placeholder")] = event.target.querySelector("input").value;
+    } else if (event.type === "input") {
+      newValue[event.target.getAttribute("placeholder")] = event.target.value;
+    }
+
+    this.setState({value: newValue});
+  }
+
+  onRelativeSelectionChange(event) {
+    var optionElement = event.target.childNodes[event.target.selectedIndex];
+    var from = moment(parseInt(optionElement.getAttribute('data-from')));
+    var to = moment(parseInt(optionElement.getAttribute('data-to')));
+    var newValue = { value: optionElement.value, from: from.format(this.props.dateFormat), to: to.format(this.props.dateFormat) };
+
+    this.setState({ value: newValue });
+    this.updateFilter(newValue);
+  }
+
+  onBlur() {
+    this.updateFilter(this.state.value);
+  }
+
+  updateFilter(newValue) {
+    this.context.filterBarActor.updateFilter(this.props.filterUid, "value", newValue);
+  }
+
+  // TODO: Update relative dates based on relative selection (if in query params) on page load, rather than applying stored dates directly
+  componentDidMount() {
+    var datePickerFrom = $(React.findDOMNode(this.refs.dateRangeFrom));
+    if (datePickerFrom.datetimepicker !== undefined) {
+      datePickerFrom.datetimepicker({ locale: 'en-au', format: 'L' });
+      datePickerFrom.datetimepicker().on("dp.change", this.onDateInputChange.bind(this));
+    }
+
+    var datePickerTo = $(React.findDOMNode(this.refs.dateRangeTo));
+    if (datePickerTo.datetimepicker !== undefined) {
+      datePickerTo.datetimepicker({ locale: 'en-au', format: 'L' });
+      datePickerTo.datetimepicker().on("dp.change", this.onDateInputChange.bind(this));
+    }
+  }
+
+  displayOptions() {
+    var lastWeek = moment().subtract(1, 'week');
+    const optionsList = [
+      { label: 'Select Period' , from: null , to: null },
+      { label: 'Today' , from: moment() , to: moment() },
+      { label: 'Last week' , from: lastWeek.clone().startOf('isoWeek'), to: lastWeek.clone().endOf('isoWeek') },
+      { label: 'This week' , from: moment().startOf('isoWeek'), to: moment().endOf('isoWeek') },
+    ]
+
+    let options = optionsList.map(function(item) {
+      return(
+        <option
+          key={item.label}
+          value={item.label}
+          data-from={item.from}
+          data-to={item.to}
+        >
+          {item.label}
+        </option>
+      )
+    })
+
+    return(
+      {options}
+    )
+  }
+
+  render() {
```

Comment:
The majority of this component is duplicating work in `<DateInput />`. '`componentDidMount ' could be dropped and this could be simplified to:
```suggestion
  render() {
    return (
      <DateInput value={this.state.value}>
        <select
          className="form-control"
          onChange={this.onRelativeChange.bind(this)}
          value={this.state.value.value}
        >
          {this.relativeOptions().map(({ label, from, to }) => (
            <option key={label} value={label} data-from={from} data-to={to}>
              {label}
            </option>
          ))}
        </select>
      </DateInput>
    )
  }
```
```